ui/src/App.e2e.test.tsx

```diff
@@ -0,0 +1,22 @@
+import puppeteer from 'puppeteer';
+import { execa } from 'execa';
+
+describe('Puppeteer E2E test', () => {
+  it('should load the webpage', async () => {
+    // Build and start the app
+    await execa('yarn', ['build']);
+    execa('yarn', ['preview'], {
+      detached: true,
+      stdio: 'ignore',
+    });
```

Comment:
Does this run indefinitely? Does something stop it?

Reply:
Good point, will investigate and get back to you. _(I can't await it or the test won't proceed.)_

Reply:
IIRC, execa returns some child process thingy than you can kill when you're done with it.

ava has a `test.shutdown()` thingy. Does this test framework?

Reply:
Thank you, I tried `subprocess.kill('SIGTERM', {...});` here with no luck:

<details>
<summary>Test Code</summary>

```js
import puppeteer from 'puppeteer';
import { execa, ExecaChildProcess } from 'execa';

let subprocess: ExecaChildProcess;

beforeAll(() => {
  subprocess = execa('node');
});

describe('Puppeteer E2E test', () => {
  it('should load the webpage', async () => {
    await execa('yarn', ['build']);
    execa('yarn', ['preview'], {
      detached: true,
      stdio: 'ignore',
    });
    // delay for preview server to start
    await new Promise(resolve => setTimeout(resolve, 2000));

    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.goto('http://localhost:4173');

    const buttonText = await page.$eval('button', el => el.textContent);
    expect(buttonText).toBe('Connect Wallet');

    await browser.close();
  });
});

afterAll(() => {
  subprocess.kill('SIGTERM', {
    forceKillAfterTimeout: 50,
  });
});
```
</details>

<details>
<summary>Test of Test Code</summary>

```bash
# success = 0 echos:

for port in {4170..4180}; do
  pid=$(lsof -ti:$port)
  if [ ! -z "$pid" ]; then
    echo "Killing process on port $port with PID $pid"
    kill -9 $pid
  fi
done
```
</details>

Went with something in d3b726c0 that seems simple enough.

Reply:
[ci failed 🫠](https://github.com/Agoric/dapp-offer-up/actions/runs/7648738949/job/20842029650?pr=57#step:8:15). Time to try & get [act](https://github.com/nektos/act) going again.

Reply:
Using `subprocess.kill()` as of [85dc90c](https://github.com/Agoric/dapp-offer-up/pull/57/commits/85dc90c021dbc7aa3bdb100727a97a943bc7f1d3)

Reply:
And no `execa` dep: [bd06653](https://github.com/Agoric/dapp-offer-up/pull/57/commits/bd0665302223be1f774573ca27d5a8eb95618293)